src/components/tableRow.jsx

```diff
@@ -49,35 +68,72 @@ class TableRow extends Component<Object> {
     const {
       data,
       cellComponent,
+      editable,
       editIndividualRows,
+      bulkEdit,
       startEditingContent,
       removeContent,
       classes,
+      updateGridState,
+      toggleSelect,
+      selectedRecords,
+      columnModel,
     } = this.props;
+
+    let isError = false;
+    if (cellComponent) {
+      columnModel.get().forEach((column) => {
+        if (column.meta) {
+          const currentValidators = column.meta.validators || [];
+          if (column.meta.required) {
+            currentValidators.push(RequiredFieldValidator);
+          }
+
+          currentValidators.forEach((validate) => {
+            if (validate(data[column.dataIndex])) {
+              isError = true;
+            }
+          });
+        }
+      });
+    }
+
     return (
-      <Table.Row>
-        {editIndividualRows && (
+      <Table.Row negative={isError}>
+        {editable && bulkEdit && (
           <Table.Cell collapsing verticalAlign="top">
-            <div className={classes.buttonWrapper}>
-              <Icon className={classes.button} link name="pencil" onClick={() => startEditingContent(data.reduxFormIndex)} />
-              <Icon className={classes.button} link name="trash" onClick={() => removeContent(data.reduxFormIndex)} />
+            <div className={classes.checkboxWrapper}>
+              <Checkbox
+                className={classes.checkbox}
+                checked={selectedRecords.includes(data.reduxFormIndex)}
```

Comment:
Can LocalStore maintain information on whether a row is check or not. That way this component does not need to be dependent on the edit plugin and call functions on it.

Reply:
I didn't put it in the LocalStore, but instead added applying a selected property to the data record, so the DataGrid component can be oblivious of the editHandler.